### ISSUE TITLE
minio server crash on ppc64le shortly after server started

### ISSUE THREAD

**Issue by shixuguang** [minio/minio#16750]
shortly after minio server started(roughly ~5min), segmentation fault  happened causing server to stop with the following error msg:

```
unexpected fault address 0xffffffffffffefa8
fatal error: fault
[signal SIGSEGV: segmentation violation code=0x3 addr=0xffffffffffffefa8 pc=0x19d462c]

goroutine 604 [running]:
runtime.throw({0x23cd33f?, 0x0?})
	runtime/panic.go:1047 +0x40 fp=0xc003e32938 sp=0xc003e328f8 pc=0x4c210
runtime.sigpanic()
	runtime/signal_unix.go:842 +0x1b8 fp=0xc003e32978 sp=0xc003e32938 pc=0x64d58
github.com/minio/minio/cmd.(*ReplicationStats).calculateBucketReplicationStats(0xc002f70000, {0xc00726e168, 0x12}, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	github.com/minio/minio/cmd/bucket-replication-stats.go:371 +0x86c fp=0xc003e41ba0 sp=0xc003e32998 pc=0x19d462c
github.com/minio/minio/cmd.(*ReplicationStats).getAllLatest(0x0?, 0xc005196720)
	github.com/minio/minio/cmd/bucket-replication-stats.go:282 +0x1c4 fp=0xc003e41e28 sp=0xc003e41ba0 pc=0x19d3b74
github.com/minio/minio/cmd.(*ReplicationPool).saveStatsToDisk(0xc0031798c0)
	github.com/minio/minio/cmd/bucket-replication.go:3076 +0x258 fp=0xc003e41f98 sp=0xc003e41e28 pc=0x19fe2c8
github.com/minio/minio/cmd.NewReplicationPool.func5()
	github.com/minio/minio/cmd/bucket-replication.go:1612 +0x4c fp=0xc003e41fc0 sp=0xc003e41f98 pc=0x19f1f6c
runtime.goexit()
	runtime/asm_ppc64x.s:902 +0x4 fp=0xc003e41fc0 sp=0xc003e41fc0 pc=0x82b64
created by github.com/minio/minio/cmd.NewReplicationPool
	github.com/minio/minio/cmd/bucket-replication.go:1612 +0x608
```
when server stands up, normal operations to/from buckets work fine

## Expected Behavior
minio server runs fine on power/ppc64le

## Current Behavior
server crash

```
unexpected fault address 0xffffffffffffefa8
fatal error: fault
[signal SIGSEGV: segmentation violation code=0x3 addr=0xffffffffffffefa8 pc=0x19d462c]

goroutine 604 [running]:
runtime.throw({0x23cd33f?, 0x0?})
	runtime/panic.go:1047 +0x40 fp=0xc003e32938 sp=0xc003e328f8 pc=0x4c210
runtime.sigpanic()
	runtime/signal_unix.go:842 +0x1b8 fp=0xc003e32978 sp=0xc003e32938 pc=0x64d58
github.com/minio/minio/cmd.(*ReplicationStats).calculateBucketReplicationStats(0xc002f70000, {0xc00726e168, 0x12}, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	github.com/minio/minio/cmd/bucket-replication-stats.go:371 +0x86c fp=0xc003e41ba0 sp=0xc003e32998 pc=0x19d462c
github.com/minio/minio/cmd.(*ReplicationStats).getAllLatest(0x0?, 0xc005196720)
	github.com/minio/minio/cmd/bucket-replication-stats.go:282 +0x1c4 fp=0xc003e41e28 sp=0xc003e41ba0 pc=0x19d3b74
github.com/minio/minio/cmd.(*ReplicationPool).saveStatsToDisk(0xc0031798c0)
	github.com/minio/minio/cmd/bucket-replication.go:3076 +0x258 fp=0xc003e41f98 sp=0xc003e41e28 pc=0x19fe2c8
github.com/minio/minio/cmd.NewReplicationPool.func5()
	github.com/minio/minio/cmd/bucket-replication.go:1612 +0x4c fp=0xc003e41fc0 sp=0xc003e41f98 pc=0x19f1f6c
runtime.goexit()
	runtime/asm_ppc64x.s:902 +0x4 fp=0xc003e41fc0 sp=0xc003e41fc0 pc=0x82b64
created by github.com/minio/minio/cmd.NewReplicationPool
	github.com/minio/minio/cmd/bucket-replication.go:1612 +0x608
```

## Possible Solution
NA

## Steps to Reproduce (for bugs)
run 4 servers on power9 architecture openshift containers
`cpu             : POWER9 (architected), altivec supported`

## Context

usability on ppc64le system

## Regression

No

## Your Environment

* Version used (`minio --version`): 
```
minio version RELEASE.2023-02-27T18-10-45Z (commit-id=0ff931dc765d5d8fdddcdf7687fec1b0b194fca9)
Runtime: go1.19.6 linux/ppc64le
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2023 MinIO, Inc.
```
* Server setup and configuration:
4 servers with one drive each

* Operating System and version (`uname -a`):
openshift 4.11.9
Linux worker0.5x-pwr.cp.fyre.ibm.com 4.18.0-372.26.1.el8_6.ppc64le #1 SMP Sat Aug 27 13:55:57 EDT 2022 ppc64le ppc64le ppc64le GNU/Linux

**Comment by klauspost** [comment]
I can't see how [this line](https://github.com/minio/minio/blob/0ff931dc765d5d8fdddcdf7687fec1b0b194fca9/cmd/bucket-replication-stats.go#L371) can panic.

* `s` is created here and isn't shared.

Looking at the neighboring lines:

* `u.ReplicaSize` is local to the function (value parameter)
* `r` is checked against nil at the top.
* `r.mostRecentStatsMu` is never written to.
* `globalReplicationStats` (where r is stored is not dereferenced to copy the lock).

This "smells" like a runtime bug.

**Comment by shixuguang** [comment]
any plan to investigate/test further or any suggestion on this issue? thanks

**Comment by harshavardhana** [comment]
> any plan to investigate/test further or any suggestion on this issue? thanks

There are no plans @shixuguang we have no access to this kind of hardware. 

If this is a serious production requirement for you feel free to reach out to us if you want this to be looked at with priority at https://min.io/pricing

**Comment by shixuguang** [comment]
there is a fix from go(runtime) on this issue https://github.com/golang/go/issues/59196#event-8841589408, and has been made available in all streams (1.20.3, 1.19.8), we've verified that the fix working for us, any plan for a release with refreshed runtime? cc:@sumitd2
thanks

**Comment by sumitd2** [comment]
@harshavardhana @klauspost This issue has been fixed by golang in 1.20.3, 1.19.8 (https://github.com/golang/go/issues/59196). Can you please cut a new release based on one of these? Thanks. @shixuguang

**Comment by sumitd2** [comment]
@shixuguang 
I just verified, the minio crash is fixed in minio.RELEASE.2023-04-13T03-08-07Z, to be downloaded from here: https://dl.min.io/server/minio/release/linux-ppc64le/. Basically minio built with Golang 1.20.3 fixes the issue.
Thank you minio team.
